flang/lib/Lower/Bridge.cpp

```diff
@@ -2292,6 +2286,26 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     localSymbols.clear();
   }
 
+  /// Instantiate the data from a BLOCK DATA unit.
+  void lowerBlockData(Fortran::lower::pft::BlockDataUnit &bdunit) {
+    // FIXME: get rid of the bogus function context and instantiate the globals
+    // directly into the module.
+    auto *context = &getMLIRContext();
+    auto func = Fortran::lower::FirOpBuilder::createFunction(
+        mlir::UnknownLoc::get(context), getModuleOp(),
+        uniquer.doGenerated("Sham"),
+        mlir::FunctionType::get(llvm::None, llvm::None, context));
+    builder = new Fortran::lower::FirOpBuilder(func, bridge.getKindMap());
```

Comment:
Is the bogus func only needed to create the builder or is there another reason it is currently needed ?

Reply:
The current lowering infrastructures also builds the variable attributes, but they have no logical Region to be dumped into here. I plan to do more refactoring so as to clean that up.